modules/gov/gov_params.go

```diff
@@ -0,0 +1,83 @@
+package gov
+
+import (
+	"encoding/json"
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+)
+
+var DepositProcedureParameter DepositProcedureParam
+
+type DepositProcedureParam struct {
+	Value DepositProcedure
+	ps    params.Setter
+	pg    params.Getter
+}
+
+func (param *DepositProcedureParam) InitGenesis(genesisState interface{}) {
+	if value, ok := genesisState.(DepositProcedure); ok {
+		param.Value = value
+	} else {
+		param.Value = DepositProcedure{
+			MinDeposit:       sdk.Coins{sdk.NewInt64Coin("iris", 10)},
+			MaxDepositPeriod: 1440}
+	}
+}
+
+func (param *DepositProcedureParam) SetReadWriter(setter params.Setter) {
+	param.ps = setter
+	param.pg = setter.Getter
+}
+
+func (param *DepositProcedureParam) GetStoreKey() string {
+	return "Gov/gov/depositProcedure"
+
+}
+
+func (param *DepositProcedureParam) SaveValue(ctx sdk.Context) {
+	param.ps.Set(ctx, param.GetStoreKey(), param.Value)
+}
+
+func (param *DepositProcedureParam) LoadValue(ctx sdk.Context) bool {
+	err := param.pg.Get(ctx, param.GetStoreKey(), &param.Value)
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func (param *DepositProcedureParam) ToJson() string {
+	jsonBytes, _ := json.Marshal(param.Value)
+	return string(jsonBytes)
+}
+
+func (param *DepositProcedureParam) Update(ctx sdk.Context, jsonStr string) {
+	if err := json.Unmarshal([]byte(jsonStr), &param.Value); err == nil {
+		param.SaveValue(ctx)
+	}
+}
+
+func (param *DepositProcedureParam) Valid(jsonStr string) sdk.Error {
+
+	var err error
+
+	if err = json.Unmarshal([]byte(jsonStr), &param.Value); err == nil {
+
+		if param.Value.MinDeposit[0].Denom != "iris" {
+			return sdk.NewError(DefaultCodespace, 102, fmt.Sprintf("It should be iris "))
```

Comment:
Should use the enum instead of the magic number 102

Reply:
It maybe need more design.